api/src/main/java/org/openmrs/Obs.java

```diff
@@ -106,55 +133,91 @@
 	 * 
 	 * @see #isObsGrouping() (??)
 	 */
+	@JoinColumn(name = "obs_group_id")
+	@ManyToOne
 	protected Obs obsGroup;
 	
 	/**
 	 * The list of obs grouped under this obs.
 	 */
 	@AllowDirectAccess
+	@JoinColumn(name = "obs_group_id")
+	@OneToMany(cascade = CascadeType.REMOVE)
+	@OrderBy("obs_id")
+	@BatchSize(size = 25)
 	protected Set<Obs> groupMembers;
 	
+	@JoinColumn(name = "value_coded")
+	@ManyToOne
 	protected Concept valueCoded;
 	
+	@JoinColumn(name = "value_coded_name_id")
+	@ManyToOne
 	protected ConceptName valueCodedName;
 	
+	@JoinColumn(name = "value_drug")
+	@ManyToOne
 	protected Drug valueDrug;
 	
+	@Column(name = "value_group_id")
 	protected Integer valueGroupId;
 	
+	@Column(name = "value_datetime", length = 19)
 	protected Date valueDatetime;
 	
+	@Column(name = "value_numeric", length = 22)
 	protected Double valueNumeric;
 	
+	@Column(name = "value_modifier", length = 2)
 	protected String valueModifier;
 	
+	@Column(name = "value_text", length = 65535)
 	protected String valueText;
 	
+	@Column(name = "value_complex")
 	protected String valueComplex;
 	
 	// ComplexData is not persisted in the database.
 	protected transient ComplexData complexData;
 	
+	@Column(name = "comments")
 	protected String comment;
 	
-	protected transient Integer personId;
+	@Column(name = "person_id", nullable = false, insertable = false, updatable = false)
+	protected Integer personId;
 	
+	@JoinColumn(name = "person_id")
+	@ManyToOne(optional = false)
 	protected Person person;
 	
+	@JoinColumn(name = "order_id")
+	@ManyToOne
 	protected Order order;
 	
+	@JoinColumn(name = "location_id")
+	@ManyToOne
 	protected Location location;
 	
+	@JoinColumn(name = "encounter_id")
+	@ManyToOne
 	protected Encounter encounter;
 	
+	@JoinColumn(name = "previous_version", unique = true)
+	@ManyToOne
 	private Obs previousVersion;
 	
+	@Column(name = "form_namespace_and_path")
+	@Access(AccessType.FIELD)
 	private String formNamespaceAndPath;
 	
-	private Boolean dirty = Boolean.FALSE;
```

Comment:
so we are having this to be persisted after these changes?

Reply:
@reagan-meant , can you squash the commits into one ?

Reply:
Currently it's @wluyima working on this issue

Reply:
Since @reagan-meant did some work on this, I think it's okay not to squash the commits to keep his commits in the history